cmake/kokkos_compiler_id.cmake

```diff
@@ -5,26 +5,29 @@ SET(KOKKOS_CXX_COMPILER_ID ${CMAKE_CXX_COMPILER_ID})
 SET(KOKKOS_CXX_COMPILER_VERSION ${CMAKE_CXX_COMPILER_VERSION})
 
 IF(Kokkos_ENABLE_CUDA)
-# Check if the compiler is nvcc (which really means nvcc_wrapper).
-EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
-                OUTPUT_VARIABLE INTERNAL_COMPILER_VERSION
-                OUTPUT_STRIP_TRAILING_WHITESPACE)
+  # Check if the compiler is nvcc (which really means nvcc_wrapper).
+  EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
+                  OUTPUT_VARIABLE INTERNAL_COMPILER_VERSION
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-STRING(REPLACE "\n" " - " INTERNAL_COMPILER_VERSION_ONE_LINE ${INTERNAL_COMPILER_VERSION} )
+  STRING(REPLACE "\n" " - " INTERNAL_COMPILER_VERSION_ONE_LINE ${INTERNAL_COMPILER_VERSION} )
 
-STRING(FIND ${INTERNAL_COMPILER_VERSION_ONE_LINE} "nvcc" INTERNAL_COMPILER_VERSION_CONTAINS_NVCC)
+  STRING(FIND ${INTERNAL_COMPILER_VERSION_ONE_LINE} "nvcc" INTERNAL_COMPILER_VERSION_CONTAINS_NVCC)
 
 
-STRING(REGEX REPLACE "^ +" ""
-       INTERNAL_HAVE_COMPILER_NVCC "${INTERNAL_HAVE_COMPILER_NVCC}")
-IF(${INTERNAL_COMPILER_VERSION_CONTAINS_NVCC} GREATER -1)
-  SET(INTERNAL_HAVE_COMPILER_NVCC true)
-ELSE()
-  SET(INTERNAL_HAVE_COMPILER_NVCC false)
-ENDIF()
+  STRING(REGEX REPLACE "^ +" ""
+         INTERNAL_HAVE_COMPILER_NVCC "${INTERNAL_HAVE_COMPILER_NVCC}")
+  IF(${INTERNAL_COMPILER_VERSION_CONTAINS_NVCC} GREATER -1)
+    SET(INTERNAL_HAVE_COMPILER_NVCC true)
+  ELSE()
+    SET(INTERNAL_HAVE_COMPILER_NVCC false)
+  ENDIF()
 ENDIF()
 
 IF(INTERNAL_HAVE_COMPILER_NVCC)
+  # Save the host compiler id before overwriting it.
+  SET(KOKKOS_CXX_HOST_COMPILER_ID ${KOKKOS_CXX_COMPILER_ID})
```

Comment:
Is this the only place the host variable gets set?

Reply:
Yes, we are only using it if the compiler is `nvcc_wrapper`. In the end, we could also replace `KOKKOS_CXX_HOST_COMPILER_ID` by `KOKKOS_CXX_COMPILER_ID` globally or only define it locally if you feel strongly about it. At this point, I am fine with doing either.